src/components/common/histogram-plot.js

```diff
@@ -0,0 +1,82 @@
+import React, {useMemo} from 'react';
+import {scaleLinear} from 'd3-scale';
+import {max} from 'd3-array';
+import styled from 'styled-components';
+import classnames from 'classnames';
+
+const histogramStyle = {
+  highlightW: 0.7,
+  unHighlightedW: 0.4
+};
+
+const HistogramWrapper = styled.svg`
+  overflow: visible;
+  .histogram-bars {
+    rect {
+      fill: ${props => props.theme.histogramFillOutRange};
+    }
+    rect.in-range {
+      fill: ${props => props.theme.histogramFillInRange};
+    }
+  }
+`;
+
+function HistogramPlotFactory() {
+  const HistogramPlot = ({width, height, margin, isRanged, histogram, value, brushComponent}) => {
+    const domain = useMemo(() => [histogram[0].x0, histogram[histogram.length - 1].x1], [
+      histogram
+    ]);
+    const dataId = Object.keys(histogram[0]).filter(k => k !== 'x0' && k !== 'x1')[0];
```

Comment:
it's very cheap operation, and the return value is a string, it's probably more expensive to memorize it than just call it